vlog4j-parser/src/main/java/org/semanticweb/vlog4j/parser/DirectiveHandler.java

```diff
@@ -0,0 +1,221 @@
+	 */
+	public static File validateFilenameArgument(final DirectiveArgument argument, final String description)
+			throws ParsingException {
+		String fileName = DirectiveHandler.validateStringArgument(argument, description);
+		File file = new File(fileName);
+		try {
+			// we don't care about the actual path, just that there is one.
+			file.getCanonicalPath();
+		} catch (IOException e) {
+			throw new ParsingException(description + "\"" + argument + "\" is not a valid file path.", e);
+		}
+
+		return file;
+	}
+
+	/**
+	 * Validate that the provided argument is an IRI.
+	 *
+	 * @param argument    the argument to validate
+	 * @param description a description of the argument, used in constructing the
+	 *                    error message.
+	 *
+	 * @throws ParsingException when the given argument is not an IRI.
+	 *
+	 * @return the contained IRI.
+	 */
+	public static URI validateIriArgument(final DirectiveArgument argument, final String description)
+			throws ParsingException {
+		try {
+			return argument.fromIri().get();
+		} catch (NoSuchElementException e) {
+			throw new ParsingException(description + "\"" + argument + "\" is not an IRI.", e);
+		}
+	}
+
+	/**
+	 * Validate that the provided argument is a {@link URL}.
+	 *
+	 * @param argument    the argument to validate
+	 * @param description a description of the argument, used in constructing the
+	 *                    error message.
+	 *
+	 * @throws ParsingException when the given argument is not a valid {@link URL}.
+	 *
+	 * @return the {@link URL} corresponding to the contained IRI.
+	 */
+	public static URL validateUrlArgument(final DirectiveArgument argument, final String description)
+			throws ParsingException {
+		URI iri = DirectiveHandler.validateIriArgument(argument, description);
+		try {
+			return iri.toURL();
+		} catch (MalformedURLException e) {
+			throw new ParsingException(description + "\"" + argument + "\" is not a valid URL.", e);
+		}
+	}
+
+	/**
+	 * Validate that the provided argument is a {@link Term}.
+	 *
+	 * @param argument    the argument to validate
+	 * @param description a description of the argument, used in constructing the
+	 *                    error message.
+	 *
+	 * @throws ParsingException when the given argument is not a {@link Term}.
+	 *
+	 * @return the contained {@link Term}.
+	 */
+	public static Term validateTermArgument(final DirectiveArgument argument, final String description)
+			throws ParsingException {
+		try {
+			return argument.fromTerm().get();
+		} catch (NoSuchElementException e) {
+			throw new ParsingException(description + "\"" + argument + "\" is not a string.", e);
+		}
+	}
+
+	/**
+	 * Obtain a {@link KnowledgeBase} from a {@link SubParserFactory}.
+	 *
+	 * @param subParserFactory the SubParserFactory.
+	 *
+	 * @return the knowledge base.
+	 */
+	default KnowledgeBase getKnowledgeBase(SubParserFactory subParserFactory) {
+		JavaCCParser subParser = subParserFactory.makeSubParser("");
```

Comment:
would it be a good idea to expose a method for this in SubParserFactory?

Reply:
I don't think so. It's really only supposed to be used in handling directives, which may affect the KnowledgeBase, but not any piece of code that gets a SubParserFactory should modify the KnowledgeBase. Even if it's still possible anyway, I feel that having such a message would encourage that.

Reply:
ok